spyder/utils/programs.py

```diff
@@ -288,7 +287,10 @@ def run_python_script_in_terminal(fname, wdir, args, interact,
                                    "an external terminal"),
                                  QMessageBox.Ok)
     elif os.name == 'posix':
-        programs = [{'cmd': 'gnome-terminal',
+        programs = [{'cmd': 'x-terminal-emulator',
+                     'wdir-option': '--working-directory',
+                     'execute-option': '-x'},

```

Comment:
It seems this change broke one of our tests. Is it really necessary to solve issue #4769?